src/index.ts

```diff
@@ -29,89 +30,98 @@ const dockerBaseUrl = "docker.io/library";
 
 async function run(): Promise<void> {
     const imageInput = core.getInput("image", { required: true });
-    const tag = core.getInput("tag") || "latest";
+    const tags = core.getInput("tags") || "latest";
+    // split tags
+    tagsList = tags.split(" ");
     const registry = core.getInput("registry", { required: true });
     const username = core.getInput("username", { required: true });
     const password = core.getInput("password", { required: true });
     const tlsVerify = core.getInput("tls-verify");
     const digestFileInput = core.getInput("digestfile");
 
-    imageToPush = `${imageInput}:${tag}`;
+    imageToPush = `${imageInput}`;
+    const registryPathList: string[] = [];
 
-    // check if image exist in Podman image storage
+    // check if image with all the required tags exist in Podman image storage
     const isPresentInPodman: boolean = await checkImageInPodman();
 
-    // check if image exist in Docker image storage and if exist pull the image to Podman
+    // check if image with all the required tags exist in Docker image storage
+    // and if exist pull the image with all the tags to Podman
     const isPresentInDocker: boolean = await pullImageFromDocker();
 
-    // failing if image is not found in Docker as well as Podman
+    // failing if image with any of the tag is not found in Docker as well as Podman
     if (!isPresentInDocker && !isPresentInPodman) {
-        throw new Error(`${imageToPush} not found in Podman local storage, or Docker local storage.`);
+        throw new Error(`All the tags of ${imageToPush} not found in Podman local storage, or Docker local storage.`);
```

Comment:
For here, let's print the actual tags list instead of jsut "all of the tags".

Ideally, it would print exactly which tags were missing, but we do that in the image checker functions, so this should be OK.

Reply:
Okay :+1: